.github/scripts/auto-label.cjs

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+module.exports = async (github, owner, repo, pullNumber) => {
+  const changedFiles = await getChangedFiles(github, owner, repo, pullNumber);
+  const changedFilePath = changedFiles.map((f) => {
+    if (f.filename.startsWith('ui/')) {
+      const [, module] = f.filename.split('/');
+      return module;
+    }
+
+    if (f.filename.startsWith('resources/'))
+      return `resources`;
+
+    if (f.filename.startsWith('util/'))
+      return `util`;
+
+    return undefined;
+  });
+  const changedModule = nonNullUnique(changedFilePath);
+
+  const s = new Set();
+  changedFiles.forEach((f) => {
+    const [, ext] = f.filename.split('.');
+    if (['js', 'ts'].includes(ext))
+      parseChangedLang(f.patch).forEach((x) => s.add(x));
+  });
+  const changedLang = Array.from(s);
+
+  const labels = [...changedModule, ...changedLang.map((v) => `💬${v.toUpperCase()}`)];
+  if (labels.length) {
+    console.log(`add ${labels} because module: ${changedModule}, lang: ${changedLang}`);
+    const res = await github.request(`POST /repos/${owner}/${repo}/issues/${pullNumber}/labels`, { labels });
+    console.log(res);
+  }
+};
+
+const getChangedFiles = async (github, owner, repo, pullNumber) => {
+  const listFilesOptions = github.pulls.listFiles.endpoint.merge({
+    owner,
+    repo,
+    // eslint-disable-next-line camelcase
+    pull_number: pullNumber,
+    // this is used by github api, nothing I can do with it.
+  });
+  const listFilesResponse = await github.paginate(listFilesOptions);
+  return listFilesResponse;
+};
+
+const parseChangedLang = (patch) => {
+  const set = new Set();
+  for (const lang of ['en', 'cn', 'de', 'ja', 'fr']) {
+    const pattern = new RegExp(`[-+].*${lang}: `);
```

Comment:
```suggestion
    const pattern = new RegExp(`^\s*[+].*(?:${lang}|'${lang}'): `);
```

* consider added lines here as adding translations
* be a little bit more strict about matching only whitespace before the language
* optional quotes